Parse census data

coconut750750 opened the pull request:
Resolves #2 
Added a Python script that takes a link and parses the census data into a Python object.

An example database object:
```
{
    "tract_id" : "17031311100", 
    "county" : "Cook County, IL",
    "rates" : { 
        "2000": {
            "03252000" : 0, 
        },
        "2010" : { 
            "03252010" : 0, 
            "03262010" : 0,
            "03292010" : 0,
            "03302010" : 0,
            "03312010" : 0,
            "04022010" : 0,
            "04052010" : 0,
            "04062010" : 0,
            "04072010" : 0,
            "04082010" : 0,
            "04092010" : 0,
            "04122010" : 0,
            "04132010" : 0.17,
            "04142010" : 0.17,
            "04152010" : 0.33,
            "04162010" : 0.33,
            "04192010" : 0.33,
            "04202010" : 0.5,
            "04212010" : 0.5,
            "04222010" : 0.5,
            "04232010" : 0.5,
            "04262010" : 0.5,
            "04272010" : 0.5
        }
    }
}
```

coconut750750 committed:
add python script to parse census data

coconut750750 committed:
add python script to parse census data

coconut750750 committed:
convert response rate to float

coconut750750 committed:
convert response rate to float

chyku commented:
[Data over different dates](https://www2.census.gov/programs-surveys/decennial/2010/program-management/4-release/mail-participation-rate/)

coconut750750 committed:
update package lock

coconut750750 committed:
add CensusResponse model

coconut750750 committed:
Merge branch 'parse_census_data' of https://github.com/hack4impact-uiuc/communityconnect-labs into parse_census_data

coconut750750 committed:
update Census Response object

coconut750750 committed:
add Census Response to models init

coconut750750 committed:
add endpoint to accept new census responses

coconut750750 committed:
update script to upload census response to databasea

coconut750750 commented:
Added a new endpoint, `/census_response`, to the backend to accept a batch of Census Response Entries to populate the database.

Updated the script to use that endpoint to save Census Responses.

chyku commented on backend/api/models/CensusResponse.py:
```diff
@@ -0,0 +1,23 @@
+class CensusResponse(Document, Mixin):
+    """CensusResponse Collection."""
+
+    response_id = StringField(required=True)
```
By responseId, do you mean the tract Id? could we make this into a number-based field?

chyku commented on backend/api/models/CensusResponse.py:
```diff
@@ -0,0 +1,23 @@
+
+    response_id = StringField(required=True)
+    name = StringField(required=True)
+    type = StringField(required=True)
```
These both should just be Tract and tract, as those are the only ones we're concerned with

chyku commented on backend/api/views/main.py:
```diff
@@ -0,0 +1,61 @@
+
+@main.route("/census_response", methods=["POST"])
+def populate_db():
+    data = request.get_json()
```
Could we put the webpage parsing into this endpoint? I think it would be pretty hard to make the webpage data into a json format

chyku reviewed (changes requested):
Great work! I'm just clearing up some things I didn't clarify earlier, and it'd be great if you could put webpage parsing directly in the endpoint (which is p sus ik but idk if it'd be great to be parsing the page into a json file that you have to copy & paste manually)

Also I'm going to try to clean up the backend stuff so that there aren't a thousand changes

coconut750750 committed:
add method to parse census data into database objects

coconut750750 committed:
add method to scrap file names from parent directory

coconut750750 committed:
update Census response database object

coconut750750 committed:
moved populate db to backend

coconut750750 committed:
combined everything into api endpoint

alanren2 committed:
yolo

coconut750750 committed:
fixed county bug

coconut750750 committed:
Merge branch 'master' into parse_census_data

chyku commented on backend/api/models/CensusResponse.py:
```diff
@@ -0,0 +1,26 @@
+    #     self.name = name
+    #     self.type = type
+    #     self.rate2000 = rate2000
+    #     self.rate2010 = rate2010
```
pls update comment

chyku commented on backend/api/views/web_scrap.py:
```diff
@@ -0,0 +1,21 @@
+if __name__ == '__main__':
+    link = "https://www2.census.gov/programs-surveys/decennial/2010/program-management/4-release/mail-participation-rate/"
+    links = extract_data_links(link)
+    print(links)
```
no newline- can u go to backend into venv and run `black .`

chyku commented on backend/api/views/populate_db.py:
```diff
@@ -0,0 +1,53 @@
+    responses = parse_census_data(link, "03252010", parse2000=True)
+
+    # result = requests.post(DB_LINK, json=[r.json() for r in responses[]])
+    # print(result.text)
```
clean up unused comments?

coconut750750 committed:
add comment for rates dictionary in census response object

caiyingshen reviewed (approved):
👻

looks good!! yay first PR!! 
(wait for kelley's approval before merging please)

chyku reviewed (changes requested):
just tiny things

coconut750750 committed:
reformatted code
